### PR TITLE
[Boosty] Add extractor

### LINKED ISSUE #5900: [boosty.to] now showing unsupported URL
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting a broken site
- [X] I've verified that I'm running yt-dlp version **2022.11.11** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/yt-dlp/yt-dlp/wiki/FAQ#video-url-contains-an-ampersand--and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

Worldwide

### Provide a description that is worded well enough to be understood

Using the test URL provided in the extractor https://github.com/yt-dlp/yt-dlp/blob/master/yt_dlp/extractor/odnoklassniki.py#L195, extraction fails and results in unsupported URL. 

This extractor was recently added in #4212

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
[debug] Command-line config: ['-vU', 'https://boosty.to/ikakprosto/posts/56cedaca-b56a-4dfd-b3ed-98c79cfa0167']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version 2022.11.11 [8b64402] (zip)
[debug] Python 3.10.5 (CPython x86_64 64bit) - Linux-5.4.0-131-generic-x86_64-with-glibc2.31 (OpenSSL 1.1.1f  31 Mar 2020, glibc 2.31)
[debug] exe versions: ffmpeg 4.2.7, ffprobe 4.2.7
[debug] Optional libraries: Crypto-3.15.0, certifi-2022.09.24, sqlite3-2.6.0
[debug] Proxy map: {}
[debug] Loaded 1723 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Latest version: 2022.11.11, Current version: 2022.11.11
yt-dlp is up to date (2022.11.11)
[debug] [generic] Extracting URL: https://boosty.to/ikakprosto/posts/56cedaca-b56a-4dfd-b3ed-98c79cfa0167
[generic] 56cedaca-b56a-4dfd-b3ed-98c79cfa0167: Downloading webpage
WARNING: [generic] Falling back on generic information extractor
[generic] 56cedaca-b56a-4dfd-b3ed-98c79cfa0167: Extracting information
[debug] Looking for embeds
ERROR: Unsupported URL: https://boosty.to/ikakprosto/posts/56cedaca-b56a-4dfd-b3ed-98c79cfa0167
Traceback (most recent call last):
  File "/usr/local/bin/yt-dlp/yt_dlp/YoutubeDL.py", line 1485, in wrapper
    return func(self, *args, **kwargs)
  File "/usr/local/bin/yt-dlp/yt_dlp/YoutubeDL.py", line 1561, in __extract_info
    ie_result = ie.extract(url)
  File "/usr/local/bin/yt-dlp/yt_dlp/extractor/common.py", line 674, in extract
    ie_result = self._real_extract(url)
  File "/usr/local/bin/yt-dlp/yt_dlp/extractor/generic.py", line 2721, in _real_extract
    raise UnsupportedError(url)
yt_dlp.utils.UnsupportedError: Unsupported URL: https://boosty.to/ikakprosto/posts/56cedaca-b56a-4dfd-b3ed-98c79cfa0167
```

### PR DESCRIPTION
**Pull request by megapro17**
### Description of your *pull request* and other information

New extractor. Looking like vk, but completely different. 

I should create another pull request for vk dash commit? I removed it from this pull request

If you have ok videoembed id, you can download video without cookies or subscription, if you have correct referrer in headers

Fixes #5900 

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Fix or improvement to an extractor (Make sure to add/update tests)
- [x] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details>

### CODE REVIEW

File: yt_dlp/extractor/boosty.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
+from yt_dlp.utils.traversal import traverse_obj
+from .common import InfoExtractor
+from ..utils import qualities, ExtractorError
+
+
+class BoostyIE(InfoExtractor):
+    _VALID_URL = r"https?://boosty.to(/[\w-]+/posts/)(?P<id>[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12})"
```
Comment (seproDev) [inline yt_dlp/extractor/boosty.py:7]: Use single quotes. Please change this everywhere.

File: yt_dlp/extractor/boosty.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+        json_data = self._search_json(
+            r'<script type="text/plain" id="initial-state">',
+            webpage, "boosty", video_id,
+        )
+        data = json_data["posts"]["postsList"]["data"]["posts"][0]
+        for i in data["data"]:
```
Comment (seproDev) [inline yt_dlp/extractor/boosty.py:32]: Are there any posts with multiple videos?

File: yt_dlp/extractor/boosty.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
+from yt_dlp.utils.traversal import traverse_obj
+from .common import InfoExtractor
+from ..utils import qualities, ExtractorError
```
Comment (seproDev) [inline yt_dlp/extractor/boosty.py:3]:
```suggestion
from .common import InfoExtractor
from ..utils import ExtractorError, qualities
from ..utils.traversal import traverse_obj
```

**Review by seproDev — CHANGES_REQUESTED** [review]

File: yt_dlp/extractor/boosty.py (hunk 2 above)
Comment (un-def) [reply yt_dlp/extractor/boosty.py:32]: Here is a post with multiple (3)  videos: https://boosty.to/maddyson/posts/0c652798-3b35-471f-8b48-a76a0b28736f

Comment (seproDev) [reply yt_dlp/extractor/boosty.py:32]: Thanks for the info and example link. In that case, the extractor should return multiple videos in a playlist and not be hardcoded for the first video.
This PR is stale, but if you (or anyone else wants) they can open a new PR to implement the extractor.

**Pull request closed without merging**